core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -266,10 +266,12 @@ private[spark] class ExecutorAllocationManager(
       // executors and inform the cluster manager to cancel the extra pending requests
       val oldNumExecutorsTarget = numExecutorsTarget
       numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
+      if (numExecutorsTarget < oldNumExecutorsTarget) {
```

Comment:
@sryza I thought about that, but wasn't sure it was necessary.

You're worried about the case where, after executing L268, we have `maxNeeded < oldNumExecutorsTarget == numExecutorsTarget == minNumExecutors`, i.e. we need fewer than the minimum, we were already at the minimum, and will continue to be at the minimum.

If we're already at the minimum, we must have actually decreased to it at some point (and therefore set `numExecutorsToAdd = 1`), or we just started, in which case `numExecutorsToAdd == 1`).

@andrewor14 sure I'll add the comment